bugsnag-android-core/src/main/java/com/bugsnag/android/DeviceData.java

```diff
@@ -253,25 +254,14 @@ private String retrieveUniqueInstallId() {
     }
 
     /**
-     * Get the free disk space on the smallest disk
+     * Get the usable disk space on internal storage's data directory
      */
-    @Nullable
-    @SuppressWarnings("deprecation") // ignore blockSizeLong suggestions for now (requires API 18)
-    private Long calculateFreeDisk() {
-        try {
-            StatFs externalStat = new StatFs(Environment.getExternalStorageDirectory().getPath());
-            long externalBytesAvailable =
-                (long) externalStat.getBlockSize() * (long) externalStat.getBlockCount();
-
-            StatFs internalStat = new StatFs(Environment.getDataDirectory().getPath());
-            long internalBytesAvailable =
-                (long) internalStat.getBlockSize() * (long) internalStat.getBlockCount();
-
-            return Math.min(internalBytesAvailable, externalBytesAvailable);
-        } catch (Exception exception) {
-            Logger.warn("Could not get freeDisk");
-        }
-        return null;
+    @SuppressLint("UsableSpace")
+    private long calculateFreeDisk() {
+        // for this specific case we want the currently usable space, not
+        // StorageManager#allocatableBytes() as the UsableSpace lint inspection suggests
```

Comment:
why?

Reply:
[`getUsableSpace()`](https://developer.android.com/reference/java/io/File#getUsableSpace()) retrieves an estimate of the currently usable amount of disk in bytes, whereas [`getAllocatableBytes()`](https://developer.android.com/reference/android/os/storage/StorageManager#getAllocatableBytes(java.util.UUID)) typically returns a larger value because it assumes the system can delete cached files.

For general application development `getAllocatableBytes()` might be more useful, but `getUsableSpace()` gives a more accurate picture of what free disk space is available at any given time.